doc/reference.rst

```diff
@@ -8,3 +8,20 @@ Sphinx-Gallery Reference
 
 .. autodata:: sphinxgallery.__version__
     :annotation: Sphinx Gallery current version
+
+.. include:: modules/generated/sphinxgallery.__version__.examples
+.. raw:: html
+
+        <div style='clear:both'></div>
```

Comment:
It is required maybe 90% because one is floating the thumbnails and one has to signalize when that ends. 10% when it is not needed is because the theme of the website is simple enough it has no conflict with the floats. The basic/default sphinx theme, which you render locally is immune. The read the docs theme, and scikit-learn where many other divs are floated need this. But it is all your choice on the theme you use.
I'm not sure if it should be added to the .example file directly. If one uses the sphinx templates for documenting modules and classes(as in scikit-learn) I guess it is better to add it manually there and once. All because it is more a theme and style instruction. If you change the CSS of sphinx-gallery and its floats, then this becomes irrelevant or more the user choice.